src/Storage/CacheItemStorage.php

```diff
@@ -72,7 +72,8 @@ public function get($key)
 
     public function set(CacheItemInterface $item)
     {
-        $this->adapter->set($item->getKey(), $item);
+        // Add ttl for buggy adapters (double time for correct cache stampede preventing algorithm)
+        $this->adapter->set($item->getKey(), $item, $this->cacheExpiresIn * 2);
```

Comment:
What is this `2` ? Can we add it to self explaining constant please.

Reply:
It simply doubles the default expire interval to not interpolate with cache stampede prevention logic (deviating expire time). We may change that to adding some offset which is greater than maximum deviating expire time. So the cached item will expire anyway even if the cache stampede logic fails or another error occurs. This will prevent polluting cache server with the items which are already expired but had not been cleared from page-cache.

Reply:
Double time it too harsh.
I tested on my machine just now, when i set time to 120 seconds `ttl` of CacheItem becomes 240, and page expires in 4 minutes instead of 2. 

Imagine a person entering an expiration time of 20 minutes, and getting 40 minutes back instead.

I am going to merge your changes in git, and then manually removing this.

Reply:
Do not remove that coz without setting `ttl` the PSR-16 cache creates perpetual records which are never expired. If you are using external PSR-16 cache like `redis` then the storage becomes bloated with entries which are never expire even if they have defined `expires` parameter. If you want the "right" solution here then we need to set `ttl` to the maximum entry lifetime (item ttl + maximum offset from cache stampede protection algorithm). You are using `log`-based math there and I recommend to replace it with a linear random with fixed maximum offset (10 seconds, for example). This way we can create PSR-16 cache entries which are exists during cache stampede algorithm execution but also get removed from cache storage if something went wrong (or if your cache depends on the release hash which is preferred way to managing cache storages). I may implement this.

Reply:
And item expiration algorithm is independent from PSR-16 cache expiration. This code allows to gracefully expire PSR-16 cache entries without harming page-cache cache stampede prevention logic.

Reply:
Let's chat on off-git. I might be missing something.
Because from my tests I saw exactly that ttl value is being multiplied by 2 , which is not an expected behavior.